core/src/main/java/nl/nn/adapterframework/receivers/Receiver.java

```diff
@@ -1937,9 +1937,6 @@ public String toString() {
 	@IbisDoc({"10", "The source of messages"})
 	public void setListener(IListener<M> newListener) {
 		listener = newListener;
-		if (StringUtils.isEmpty(listener.getName())) {
-			listener.setName("listener of ["+getName()+"]");
-		}
```

Comment:
This sets the name of the listener, only when it's not explicitly set. I think we should keep this

Reply:
this breaks the logic. we have propagateName method which updates the name of the listener in case the name of the listener is empty.
If this code block stays then the listeners may have name like 'listener of [null]' in case the receiver does not have a name